lib/dal/src/attribute/value.rs

```diff
@@ -2125,4 +2128,108 @@ impl AttributeValue {
 
         Ok(input_socket_ids)
     }
+    #[instrument(level = "info", skip_all)]
+    /// Get the morale equivalent of the prop path for a given attribute value id.
+    /// This includes the key/index in the path, unlike the path for a Prop which doesn't
+    /// include the key/index
+    pub async fn get_path_for_id(
+        ctx: &DalContext,
+        attribute_value_id: AttributeValueId,
+    ) -> AttributeValueResult<Option<String>> {
+        let mut parts = VecDeque::new();
+        let mut work_queue = VecDeque::from([attribute_value_id]);
+
+        while let Some(mut attribute_value_id) = work_queue.pop_front() {
+            let prop_name = match AttributeValue::prop_id_for_id(ctx, attribute_value_id).await {
+                Ok(prop) => Prop::get_by_id(ctx, prop).await?.name,
+                Err(_) => {
+                    // This means it's an input or output socket, which has no path and should just be empty string
+                    continue;
+                }
+            };
```

Comment:
Is the only error possible here the one that you listed? If not, then we should handle the specific error or refactor the `prop_id_for_id` function to return a different type to help you determine that (e.g. `Result<Option<T>, E>>` instead of `Result<T, E>`).